medallion/test/test_memory_backend.py

```diff
@@ -536,15 +536,19 @@ def test_saving_data_file(self):  # just for the memory backend
         post_header["Content-Type"] = MEDIA_TYPE_TAXII_V21
         post_header["Accept"] = MEDIA_TYPE_TAXII_V21
 
-        with tempfile.NamedTemporaryFile() as f:
+        with tempfile.NamedTemporaryFile(mode="w") as f:
             self.client.post(
                 test.ADD_OBJECTS_EP,
                 data=json.dumps(new_bundle),
                 headers=post_header,
             )
-            self.app.medallion_backend.save_data_to_file(f.name)
+            self.app.medallion_backend.save_data_to_file(f)
             assert os.path.isfile(f.name)
 
+            # seek(0) is needed to reset the file pointer so it's correctly
+            # read by the backend again
+            f.seek(0)
```

Comment:
The file was opened for writing; you mustn't try to read from it.  I think you probably want mode="w+".